third_party/move/move-compiler-v2/src/pipeline/ability_processor.rs

```diff
@@ -0,0 +1,656 @@
+
+// =================================================================================================
+// Instrumenter
+
+/// Represents run for one function.
+struct Instrumenter<'a> {
+    /// A function data builder which owns the `FunctionData` which is worked on.
+    builder: FunctionDataBuilder<'a>,
+    /// The live-var information for the function.
+    live_var: &'a LiveVarAnnotation,
+    /// The result of the copy-drop analysis
+    copy_drop: BTreeMap<CodeOffset, CopyDropState>,
+}
+
+impl<'a> Instrumenter<'a> {
+    fn run(&mut self, code: Vec<Bytecode>) {
+        for (offset, bc) in code.into_iter().enumerate() {
+            self.instrument_bytecode(offset as CodeOffset, bc)
+        }
+    }
+
+    /// Instruments and checks a bytecode. See the file documentation for an overview
+    /// of what this function does.
+    fn instrument_bytecode(&mut self, code_offset: CodeOffset, bc: Bytecode) {
+        use Bytecode::*;
+        // Instrument and check bytecode
+        match bc.clone() {
+            Assign(id, dst, src, kind) => match kind {
+                AssignKind::Inferred => {
+                    let copy_drop_at = self.copy_drop.get(&code_offset).expect("copy_drop");
+                    if copy_drop_at.needs_copy.contains(&src) {
+                        self.check_implicit_copy(code_offset, id, src);
+                        self.builder.emit(Assign(id, dst, src, AssignKind::Copy))
+                    } else {
+                        self.builder.emit(Assign(id, dst, src, AssignKind::Move))
+                    }
+                },
+                AssignKind::Copy | AssignKind::Store => {
+                    self.check_explicit_copy(id, src);
+                    self.builder.emit(Assign(id, dst, src, AssignKind::Copy))
+                },
+                AssignKind::Move => {
+                    self.check_explicit_move(code_offset, id, src);
+                    self.builder.emit(Assign(id, dst, src, AssignKind::Move))
+                },
+            },
+            Call(id, dests, op, srcs, ai) => {
+                use Operation::*;
+                match &op {
+                    Function(mod_id, fun_id, insts) => {
+                        self.check_fun_inst(id, *mod_id, *fun_id, insts);
+                        let new_srcs = self.copy_arg_if_needed(code_offset, id, srcs);
+                        self.check_and_emit_bytecode(code_offset, Call(id, dests, op, new_srcs, ai))
+                    },
+                    _ => self.check_and_emit_bytecode(code_offset, Call(id, dests, op, srcs, ai)),
+                }
+            },
+            _ => self.check_and_emit_bytecode(code_offset, bc.clone()),
+        }
+        // Insert any drops needed after this program point
+        self.add_implicit_drops(code_offset, &bc)
+    }
+
+    fn check_and_emit_bytecode(&mut self, _code_offset: CodeOffset, bc: Bytecode) {
+        use Bytecode::*;
+        #[allow(clippy::single_match)]
+        match &bc {
```

Comment:
Gotcha. I think it would make sense to globally disable if we are frequently adding such exceptions.

I have not tested this, but from reading the lint rule (https://rust-lang.github.io/rust-clippy/master/#/single_match), it seems like if you add any comment in the other match arm (_), then the lint wont complain. That might be another way to go.